src/core/components/radio/index.tsx

```diff
@@ -188,11 +203,16 @@ const Radio = ({
 
 const radioGroupDefaultProps = {
 	orientation: "vertical",
+	optional: false,
 	hideLabel: false,
 }
 
 const radioDefaultProps = {
 	disabled: false,
+	type: "radio",
+	optional: false,
+	defaultChecked: false,
```

Comment:
I think having a `defaultChecked` attribute passed by default is going to cause problems when a developer also passes a `checked` attribute. React doesn't allow both attributes to be passed at the same time.